pkg/history/populate_headers_test.go

```diff
@@ -59,9 +60,31 @@ var _ = Describe("Populating headers", func() {
 	It("returns early if the db is already synced up to the head of the chain", func() {
 		blockChain := fakes.NewMockBlockChain()
 		blockChain.SetLastBlock(big.NewInt(2))
-		headersAdded, err := history.PopulateMissingHeaders(blockChain, headerRepository, 2)
+		headersAdded, err := history.PopulateMissingHeaders(blockChain, headerRepository, 2, &statusWriter)
 
 		Expect(err).NotTo(HaveOccurred())
 		Expect(headersAdded).To(Equal(0))
 	})
+
+	It("Writes a healthcheck file", func() {
+		blockChain := fakes.NewMockBlockChain()
+		blockChain.SetLastBlock(big.NewInt(2))
+		headerRepository.SetMissingBlockNumbers([]int64{2})
+
+		_, err := history.PopulateMissingHeaders(blockChain, headerRepository, 1, &statusWriter)
+
+		Expect(err).NotTo(HaveOccurred())
+		Expect(statusWriter.WriteCalled).To(BeTrue())
+	})
+
+	It("Does not write a healthcheck file when the call to get the last block fails", func() {
+		blockChain := fakes.NewMockBlockChain()
+		blockChain.SetLastBlockError(fakes.FakeError)
+
+		_, err := history.PopulateMissingHeaders(blockChain, headerRepository, 1, &statusWriter)
+
+		Expect(err).To(HaveOccurred())
+		Expect(err.Error()).To(MatchRegexp("error getting last block: failed"))
```

Comment:
I'm seeing an error when I use `MatchError` because we're wrapping the error before returning it:
```

  Expected
      <*errors.errorString | 0xc00030a6c0>: {
          s: "error getting last block: failed",
      }
  to match error
      <*errors.errorString | 0xc000072330>: {s: "failed"}
```

I feel like there's probably a better way to handle this sort of thing, but I haven't been able to figure it out.

Reply:
oh yeah I think we would need to use `%w` instead of `%s` in `fmt.Errorf` and format `err` rather than `err.Error()` in order to get a wrapped error - but then `MatchError` would match it

Reply:
Gotcha! Nice!